### ISSUE TITLE
false positive in a simple function react/destructuring-assignment

### ISSUE THREAD

**Issue by danosaure** [yannickcr/eslint-plugin-react#3024]
I have the following code:

```javascript
import fn from '../some-file';
export default (fileName) => {
  const match = fileName.match(/some expression/);
  if (match) {
    return fn;
  }
  return null;
};
```

it thinks that `const match = fileName.match()` is causing a destructuring issue.

**Comment by ljharb** [comment]
This certainly shouldn't be detected as a component, since it has no hooks, and no jsx, and the argument isn't named "props".
